### ISSUE TITLE
GPIO-STM32F051 AF correction

### ISSUE THREAD

**Issue by Bytegrity** [STMicroelectronics/STM32_open_pin_data#8]
Hi there,

While performing some file generation operations on a [repo](https://github.com/zephyrproject-rtos/hal_stm32) reliant on this one. I noticed differences caused by a changed alternate function index. I managed to trace down the source causing the unexpected output to https://github.com/STMicroelectronics/STM32_open_pin_data/blob/6f7f7a0a4d6ef31b94a63b85932db17118499ea0/mcu/IP/GPIO-STM32F051_gpio_v1_0_Modes.xml#L2251

I verified a couple of related MCU datasheets and I believe the change below may be required:
```
--- a/mcu/IP/GPIO-STM32F051_gpio_v1_0_Modes.xml
+++ b/mcu/IP/GPIO-STM32F051_gpio_v1_0_Modes.xml
@@ -2248,7 +2248,7 @@
         </SpecificParameter>
         <PinSignal Name="TIM3_CH3">
             <SpecificParameter Name="GPIO_AF">
-                <PossibleValue>GPIO_AF1_TIM3</PossibleValue>
+                <PossibleValue>GPIO_AF0_TIM3</PossibleValue>
             </SpecificParameter>
         </PinSignal>
     </GPIO_Pin>
```
After this change and running the generator in the other repo, the output line up.

**Comment by CMOSTM** [comment]
Hi

I reported the issue internally. We will have a look.
Thanks

CMO

**Comment by Bytegrity** [comment]
Hi,

Any updates on the internal discussion?

**Comment by CMOSTM** [comment]
A ticket has been raised internally to do the analysis
I will inform you about the progress of this ticket.

BR

CMO

**Comment by paaspaas00** [comment]
@CMOSTM any news?

**Comment by CMOSTM** [comment]
Hi 
It should be fixed in the next release I hope
CMO

**Comment by CMOSTM** [comment]
Fixed with the new 2024 delivery MX 6.11

**Comment by CMOSTM** [comment]
Fixed with the new 2024 delivery MX 6.11
